library/oneview_logical_interconnect_group.py

```diff
@@ -201,11 +200,17 @@ def __uplink_set_update(self):
                 allUplinkSets = self.__get_all_uplink_sets()
                 for uplinkSet in self.data['uplinkSets']:
                     networkNames = uplinkSet.pop('networkNames', None)
+                    networkSetNames = uplinkSet.pop('networkSetNames', None)
                     if networkNames and not uplinkSet.get('networkUris'):
                         uplinkSet['networkUris'] = []
                     if networkNames:
                         networkUris = [self.__get_network_uri(x) for x in networkNames]
                         uplinkSet['networkUris'].extend(networkUris)
+                    if networkSetNames and not uplinkSet.get('networkSetUris'):
+                        uplinkSet['networkSetUris'] = []
+                    if networkSetNames:
+                        networkSetUris = [self.__get_network_set(x) for x in networkSetNames]
+                        uplinkSet['networkSetUris'].extend(networkSetUris)
```

Comment:
What if there are duplicate values in networkSetUris list..? take care of this below as well

Reply:
Thats been taken care at oneview end itself. 
Tested it. 
Works fine for duplicate entry for update and for create as well.

Reply:
ok